src/components/weather-page/SingleDayDetail.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import {SingleDayData} from './SingleDayData';
+import './SingleDayDetail.scss';
+
+import ThermometerIcon from './resources/svg_icon_thermometer';
+import CloudIcon from './resources/svg_icon_cloud';
+import SunIcon from './resources/svg_icon_sun';
+import WindIcon from './resources/svg_icon_wind';
+
+export default function SingleDayDetail(props : SingleDayData) {  
+    
+    let min = props.minTemp ? Math.round(props.minTemp) + " °C" : 0 + " °C";   
+    let max = props.maxTemp ? Math.round(props.maxTemp) + " °C": 0 + " °C"; 
```

Comment:
little bit of duplicated code here - also not sure that `0` is a good option for no data, as it looks the same as actually having a temp of 0.

How about

```ts
function formatTemperature(temperature?: number): string {
    if (!number) {
        return "N/A";
    }
    return `${Math.round(temperature)} °C`;
}

const min = formatTemperature(props.minTemp);
const max = formatTemperature(props.maxTemp);
```

Reply:
We can move `formatTemperature` into its own helper function too, and add some unit tests for it.

Reply:
Thanks Mike, does the !number know the variable name by magic (genuine question) or is that a typo?

Reply:
No worries either way, I wasn't sure on the "$" syntax either so went with 

```

export default function formatTemperature (temperature?: number) : string{
    return temperature ? Math.round(temperature) + ' °C' : "N/A";
}

````

Reply:
`number` was just a typo.

the `$` syntax is [JS string literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)